### PR TITLE
Tweaking Up SpringBoot configuration

### PR DESCRIPTION
**Pull request by jazzinjars**
- Use Undertow instead of default Tomcat to improve performance
- Enabling Response Compression
- Enabling HTTP/2 support
- Enabling caching control for resources

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -120,6 +132,16 @@
             <artifactId>jaxb-api</artifactId>
             <version>2.3.0</version>
         </dependency>
+        <dependency>
```
Comment (alfonsserra) [inline pom.xml]: Do we really need the following two dependencies? I have run the application without them and works well.

Comment (jazzinjars) [reply pom.xml]: On a windows environment, the maven package goal doesn't include theses dependencies on the final jar. So, at runtime, we have a ClassNotFoundException. To solve this, I have had to add it explicitly. 
But, we should find a way to guarantee that the environment doesn't affect in the maven lifecycle. Does anyone else have this problem under a Windows environment?
I can remove these dependencies from pom file and push it again without it.
Thx

Comment (alfonsserra) [reply pom.xml]: Thank you @jazzinjars. I understand the reason. The for me is fine. Great PR

**Comment by jazzinjars** [comment]
Once we have checked the issue under a clean Windows environment, we have rolled back the addition of the dependencies. Thx @quimnoto for your support.

**Pull request merged**